torch/_C/__init__.pyi.in

```diff
@@ -2081,12 +2081,12 @@ class _CudaStreamBase(Stream):
     priority: _int
 
     def __new__(
-        self,
+        cls,
         priority: _int = 0,
         stream_id: _int = 0,
         device_index: _int = 0,
         stream_ptr: _int = 0,
-    ) -> _CudaStreamBase: ...
+    ) -> Self: ...
```

Comment:
I don't think Self is defined on __new__ so this TypeVar will be an unspecialized TypeVar.

Reply:
Self as return type is valid as long as _CudaStreamBase is not a metaclass, see (here)[https://typing.readthedocs.io/en/latest/spec/generics.html#self]